### PR TITLE
add document for metadata validator.

### PR DESCRIPTION
**Pull request by vwxyzh**
@chenkennt @superyyrrzz @hellosnow @ansyral @qinezh

### CODE REVIEW

File: Documentation/tutorial/validate your markdown files.md [hunk 1]
```diff
@@ -150,3 +150,33 @@ public class MyMarkdownTokenValidatorProvider : IMarkdownTokenValidatorProvider
...
+1.  yaml header in markdown.
+2.  global metadata.
+3.  file metadata.
+
+### Create validation plug-ins
+1.  Create a project in your code editor (e.g. visual studio).
+2.  Add nuget package `Microsoft.DocAsCode.Plugins` and `Microsoft.Composition`.
+3.  Create a class and implements @Microsoft.DocAsCode.Plugins.IInputMetadataValidator
+
+For example, we don't allow any metadata with name `hello`:
```
Comment (superyyrrzz) [inline Documentation/tutorial/validate your markdown files.md]: we don't allow
change to: 
the following validator prohibits?

**Comment by superyyrrzz** [comment]
:shipit:

File: Documentation/tutorial/validate your markdown files.md [hunk 2]
```diff
@@ -150,3 +150,33 @@ public class MyMarkdownTokenValidatorProvider : IMarkdownTokenValidatorProvider
 ## How to enable custom validators
 1.  Config markdown validation rules in `md.style`.
 2.  Copy your extension assemblies to `plugin` folder in your `DocFX.exe` or template.
+
+## Validate metadata in markdown files
+In markdown file, we can write some metadata in [conceptual](../spec/docfx_flavored_markdown.md#yaml-header) or [overwrite document](intro_overwrite_files.md)
+And we allow add some plug-ins to validate metadata write in markdown files.
+
+### Scope of validation
+Following metadata will be validate:
```
Comment (hellosnow) [inline Documentation/tutorial/validate your markdown files.md]: validated

File: Documentation/tutorial/validate your markdown files.md [hunk 3]
```diff
@@ -150,3 +150,33 @@ public class MyMarkdownTokenValidatorProvider : IMarkdownTokenValidatorProvider
 ## How to enable custom validators
 1.  Config markdown validation rules in `md.style`.
 2.  Copy your extension assemblies to `plugin` folder in your `DocFX.exe` or template.
+
+## Validate metadata in markdown files
+In markdown file, we can write some metadata in [conceptual](../spec/docfx_flavored_markdown.md#yaml-header) or [overwrite document](intro_overwrite_files.md)
+And we allow add some plug-ins to validate metadata write in markdown files.
```
Comment (hellosnow) [inline Documentation/tutorial/validate your markdown files.md]: written in

File: Documentation/tutorial/validate your markdown files.md [hunk 4]
```diff
@@ -150,3 +150,33 @@ public class MyMarkdownTokenValidatorProvider : IMarkdownTokenValidatorProvider
...
+### Scope of validation
+Following metadata will be validate:
+1.  yaml header in markdown.
+2.  global metadata.
+3.  file metadata.
+
+### Create validation plug-ins
+1.  Create a project in your code editor (e.g. visual studio).
+2.  Add nuget package `Microsoft.DocAsCode.Plugins` and `Microsoft.Composition`.
+3.  Create a class and implements @Microsoft.DocAsCode.Plugins.IInputMetadataValidator
```
Comment (hellosnow) [inline Documentation/tutorial/validate your markdown files.md]: implement?

**Comment by hellosnow** [comment]
:shipit:

File: Documentation/tutorial/validate your markdown files.md (hunk 3 above)
Comment (qinezh) [inline Documentation/tutorial/validate your markdown files.md]: usage?

**Comment by qinezh** [comment]
:shipit:

File: Documentation/tutorial/validate your markdown files.md (hunk 1 above)
Comment (vwxyzh) [inline Documentation/tutorial/validate your markdown files.md]: fixed.

File: Documentation/tutorial/validate your markdown files.md (hunk 4 above)
Comment (vwxyzh) [inline Documentation/tutorial/validate your markdown files.md]: fixed.

File: Documentation/tutorial/validate your markdown files.md (hunk 2 above)
Comment (vwxyzh) [inline Documentation/tutorial/validate your markdown files.md]: fixed.

File: Documentation/tutorial/validate your markdown files.md (hunk 3 above)
Comment (vwxyzh) [inline Documentation/tutorial/validate your markdown files.md]: fixed.

File: Documentation/tutorial/validate your markdown files.md [hunk 5]
```diff
@@ -150,3 +150,33 @@ public class MyMarkdownTokenValidatorProvider : IMarkdownTokenValidatorProvider
...
+3.  Create a class and implement @Microsoft.DocAsCode.Plugins.IInputMetadataValidator
+
+For example, the following validator prohibits any metadata with name `hello`:
+```csharp
+[Export(typeof(IInputMetadataValidator))]
+public class MyInputMetadataValidator : IInputMetadataValidator
+{
+    public void Validate(string sourceFile, ImmutableDictionary<string, object> metadata)
+    {
+        if (metadata.ContainKey("hello"))
```
Comment (hexiaokai) [inline Documentation/tutorial/validate your markdown files.md]: ContainsKey

File: Documentation/tutorial/validate your markdown files.md [hunk 6]
```diff
@@ -150,3 +150,33 @@ public class MyMarkdownTokenValidatorProvider : IMarkdownTokenValidatorProvider
 ## How to enable custom validators
 1.  Config markdown validation rules in `md.style`.
 2.  Copy your extension assemblies to `plugin` folder in your `DocFX.exe` or template.
+
+## Validate metadata in markdown files
```
Comment (hexiaokai) [inline Documentation/tutorial/validate your markdown files.md]: I think this section should be placed before the ##How to enable section. and need to let user know that to enable this, the md.style file doesn't need to be configured..

**Pull request merged**